### PR TITLE
Pd contribs personal details v2

### PR DESCRIPTION
**Pull request by paul-daniel-dempsey**
We need a component for collecting the user's email address and- if they are making a recurring contribution- their first and last names.
It should have an accompanying container component that can update the personalDetails Redux slice when those details are changed.

If picking up this card before the design specs are ready, we can go with a rough implementation and then revisit the UI later

https://trello.com/c/1XhAb50x/757-v2-checkout-build-personal-details-component

## Why are you doing this?
Requirements

1 Single Contribution
1.1 Signed In
  Email is pre-populated
  Sign out link rendered
Screenshot 2022-10-24 at 16 16 19

1.2 Signed Out
  Email is empty
  Sign out link is not rendered
  Screenshot 2022-10-24 at 14 40 39

2 Recurring Contribution
  2.1 Signed In
    2.1.1 All Regions
    Email, First Name and Last Name are pre-populated
    Sign out link rendered
Screenshot 2022-10-24 at 16 28 05

  2.1.2 US Region ONLY
  State dropdown rendered but not pre-populated
Screenshot 2022-10-24 at 16 29 27

  2.2 Signed Out
    2.2.1 All Regions
    Email, First Name and Last Name are empty
    Sign out link is not rendered
    Screenshot 2022-10-24 at 14 46 18

2.2.2 US Region
State dropdown rendered but not pre-populated
Screenshot 2022-10-24 at 14 43 33

## Is this an AB test?
- [x] Yes
#ab-supporterPlus=variant
i.e.
https://support.code.dev-theguardian.com/us/contribute#ab-supporterPlus=variant
https://support.gulocal.com/us/contribute#ab-supporterPlus=variant

If this is an AB test, PR reviewers should open and check the Optimize test.
[**Optimize Link**](https://optimize.google.com/optimize/home)

### CODE REVIEW

File: support-frontend/assets/components/personalDetails/personalDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+		lastName,
+		email,
+		isSignedIn: isSignedInPersonalDetails,
+		userTypeFromIdentityResponse,
+	});
+
+	const getFormFieldError = (formField: string) =>
+		checkoutFormHasBeenSubmitted
+			? firstError(formField, formErrors)
+			: undefined;
```
Comment (i-hardy) [inline support-frontend/assets/components/personalDetails/personalDetails.tsx:50]: We'll be handling form validation through a different, Redux-based mechanism for the new checkout, so shouldn't need the error code in this component.

File: support-frontend/assets/components/personalDetails/personalDetailsContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	renderPersonalDetails,
+}: PersonalDetailsContainerProps): JSX.Element {
+	const dispatch = useContributionsDispatch();
+
+	const { email, firstName, lastName } = useContributionsSelector(
+		(state) => state.page.checkoutForm.personalDetails,
+	);
+	const contributionType = getContributionType(
+		useContributionsSelector((state) => state),
+	);
```
Comment (i-hardy) [inline support-frontend/assets/components/personalDetails/personalDetailsContainer.tsx:31]:
```suggestion
	const contributionType = useContributionsSelector(getContributionType);
```

**Review by i-hardy — APPROVED** [review]
Couple of small changes, but otherwise this is looking great 👏 it's especially good to have stories covering all the different variations so we've got good visual regression test coverage from the start.

File: support-frontend/assets/components/personalDetails/personalDetails.tsx (hunk 1 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/personalDetails/personalDetails.tsx:50]: Ah ok, do I pass any errors generated up to the container (eg TextInput error) and handle there? Is there an example anywhere I can have a look at?

Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/personalDetails/personalDetails.tsx:50]: Actually, think understand, just remove all the associated error handling + props for the moment. Will do now, thanks

Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/personalDetails/personalDetails.tsx]: Ok, all removed

**Pull request merged**